test/cases.py

```diff
@@ -20,9 +22,22 @@
 )
 from torch.utils.data import DataLoader, TensorDataset
 
+from collections import UserDict
+
 DEVICES = get_available_devices()
 DEVICES_IDS = [f"dev={d}" for d in DEVICES]
 
+
+class ModelWithDictInput(Module):
+    def __init__(self, num_classes=2):
+        super().__init__()
+        self.net = Sequential(Linear(10, 5), ReLU(), Linear(5, num_classes))
+
+    def forward(self, data: UserDict | dict):
+        x = data["x"]
```

Comment:
When `torch.cuda.is_available()` this model has to transfer to device here, otherwise the test will fail I think.